test/medium-client-unit-tests.js

```diff
@@ -0,0 +1,115 @@
+var config = require('./config')
+var medium = require("../")
+var should = require("should")
+var testHelper = require('./test-helper')
+
+var shouldFailWithMissingParamError = function (obj, func, options, opt_requiredParam) {
+  // this allows the options to already be missing the required parameter,
+  // or have it be removed here
+  if (opt_requiredParam) {
+    delete options[opt_requiredParam]
+  }
+  (function () {
+    func.call(obj, options, function (err, data) {
+      if (err) throw err
+    })
+  }).should.throw(/^Missing required parameter/)
```

Comment:
As I decided to remove the new errror code, tests are now testing or MediumError to make sure other errors potentially thrown by the code don't give false positives